src/clusterfuzz/_internal/platforms/android/fetch_artifact.py

```diff
@@ -118,10 +118,19 @@ def download_artifact(client, bid, target, attempt_id, name, output_directory,
   return output_path
 
 
-def get_artifacts_for_build(client, bid, target, attempt_id='latest'):
+def get_artifacts_for_build(client, bid, target, attempt_id='latest',
+                            regexp=''):
   """Return list of artifacts for a given build."""
-  request = client.buildartifact().list(
-      buildId=bid, target=target, attemptId=attempt_id)
+  if regexp == '':
+    request = client.buildartifact().list(
+        buildId=bid, target=target, attemptId=attempt_id)
+  else:
+    request = client.buildartifact().list(
+        buildId=bid,
+        target=target,
+        attemptId=attempt_id,
+        nameRegexp=regexp,
+        maxResults=100)
```

Comment:
Why is maxResults passed here but not above?

Reply:
The maxResults=100 parameter is passed in the else block to ensure that all artifacts matching the provided regular expression are retrieved in a single request. This is because the default maximum result limit of 9 artifacts might be insufficient when using a regex that matches a large number of artifacts.
In earlier approach(if-case) , there were no regex based filtering and we were fetching artifacts name one by one using buildartifact.list() api resulting in more number of api calls. and no need of passing 'maxResults' argument.